diff -uNr binutils-2.20.org/gas/config/tc-v810.c binutils-2.20/gas/config/tc-v810.c
--- binutils-2.20.org/gas/config/tc-v810.c	2010-11-22 16:22:32.000000000 -0600
+++ binutils-2.20/gas/config/tc-v810.c	2010-11-22 16:21:11.000000000 -0600
@@ -463,6 +463,7 @@
 /* This table is sorted.  Suitable for searching by a binary search.  */
 static const struct reg_name pre_defined_registers[] =
 {
+  { "ep",  30 },		/* ep - element ptr.  */
   { "gp",   4 },		/* gp - global ptr.  */
   { "hp",   2 },		/* hp - handler stack ptr.  */
   { "lp",  31 },		/* lp - link ptr.  */
