diff -uNr gcc-4.4.2.org/gcc/config/v810/v810.h gcc-4.4.2/gcc/config/v810/v810.h
--- gcc-4.4.2.org/gcc/config/v810/v810.h	2010-11-23 20:08:45.000000000 -0600
+++ gcc-4.4.2/gcc/config/v810/v810.h	2010-11-23 20:07:28.000000000 -0600
@@ -487,7 +487,7 @@
    Do not define this macro if it would be the same as
    `FRAME_POINTER_REGNUM'.  */
 #undef  HARD_FRAME_POINTER_REGNUM 
-#define HARD_FRAME_POINTER_REGNUM 29
+#define HARD_FRAME_POINTER_REGNUM 25
 
 /* Base register for access to arguments of the function.  */
 #define ARG_POINTER_REGNUM 33
